cdcsdk-testing/src/test/java/com/yugabyte/cdcsdk/testing/PubSubConsumerIT.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package com.yugabyte.cdcsdk.testing;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.awaitility.Awaitility;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.testcontainers.containers.BindMode;
+
+import com.google.cloud.pubsub.v1.AckReplyConsumer;
+import com.google.cloud.pubsub.v1.MessageReceiver;
+import com.google.cloud.pubsub.v1.Subscriber;
+import com.google.pubsub.v1.ProjectSubscriptionName;
+import com.google.pubsub.v1.PubsubMessage;
+import com.yugabyte.cdcsdk.testing.util.CdcsdkTestBase;
+import com.yugabyte.cdcsdk.testing.util.UtilStrings;
+
+/**
+ * Release test that verifies inserts, updates and deletes on a YugabyteDB
+ * database and
+ * writing to Google Pub/Sub
+ *
+ * @author Sumukh Phalgaonkar
+ */
+
+public class PubSubConsumerIT extends CdcsdkTestBase {
+    private static String projectId = System.getenv("GCLOUD_PROJECT");// We are using project id ="yugabyte";
+    private static String subscriptionId = System.getenv("SUBSCRIPTION_ID");// We are using subscription id"dbserver1.public.test_table-sub";
+
+    @BeforeAll
+    public static void beforeClass() throws Exception {
+        initializeContainers();
+
+        initHelpers(true, false, false);
+    }
+
+    @BeforeEach
+    public void beforeEachTest() throws Exception {
+        ybHelper.execute(UtilStrings.getCreateTableYBStmt(DEFAULT_TABLE_NAME));
+    }
+
+    @AfterEach
+    public void dropTable() throws Exception {
+        ybHelper.execute(UtilStrings.getDropTableStmt(DEFAULT_TABLE_NAME));
+    }
+
+    @Test
+    public void automationOfPubSubAssertions() throws Exception {
+
+        cdcsdkContainer = TestHelper
+                .getCdcsdkContainerForPubSubSink(ybHelper, projectId, "public." + DEFAULT_TABLE_NAME)
+                .withCreateContainerCmdModifier(cmd -> cmd.withUser(System.getenv("USERID")));
+        cdcsdkContainer.withNetwork(containerNetwork);
+
+        cdcsdkContainer.withFileSystemBind(System.getenv("GOOGLE_APPLICATION_CREDENTIALS"),
+                "/tmp/keys/application_default_credentials.json", BindMode.READ_ONLY);
+        cdcsdkContainer.addEnv("GOOGLE_APPLICATION_CREDENTIALS", "/tmp/keys/application_default_credentials.json");
+        cdcsdkContainer.start();
+
+        assertTrue(cdcsdkContainer.isRunning());
```

Comment:
Can you move the code up to this point to `beforeEachTest` ?

Reply:
Done